File: frontend/src/models/Meeting.ts
import {StudentWork} from './StudentWork';
import {Member} from './Member';
import {CriteriaGroup} from './CriteriaGroup';

/**
 * Interface for meeting.
 *
 * @param id - Meeting id
 * @param dateAndTime - Meeting date and time
 * @param auditorium - Meeting auditorium
 * @param info - Meeting info
 * @param callLink - Meeting callLink
 * @param materialsLink - Meeting materialsLink
 * @param studentWorks - List of meeting student works
 * @param members - List of meeting memebers
 * @param criteriaGroup - Meeting criteria group
 */
export interface Meeting {
    id?: number,
    dateAndTime: Date,
    auditorium: string,
    info: string,
    callLink: string,
    materialsLink: string,
    studentWorks: StudentWork[],
    members: Member[],
    criteriaGroup?: CriteriaGroup
}

File: PracticeGrading.Data/Entities/Meeting.cs
// <copyright file="Meeting.cs" company="Maria Myasnikova">
// Copyright (c) Maria Myasnikova. All rights reserved.
// Licensed under the Apache-2.0 license. See LICENSE file in the project root for full license information.
// </copyright>

namespace PracticeGrading.Data.Entities;

/// <summary>
/// Meeting entity.
/// </summary>
public class Meeting
{
    /// <summary>
    /// Gets or sets id.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Gets or sets date and time.
    /// </summary>
    public DateTime DateAndTime { get; set; }

    /// <summary>
    /// Gets or sets auditorium.
    /// </summary>
    public string? Auditorium { get; set; }

    /// <summary>
    /// Gets or sets info.
    /// </summary>
    public string? Info { get; set; }

    /// <summary>
    /// Gets or sets call link.
    /// </summary>
    public string? CallLink { get; set; }

    /// <summary>
    /// Gets or sets materials link.
    /// </summary>
    public string? MaterialsLink { get; set; }

    /// <summary>
    /// Gets or sets student works.
    /// </summary>
    public required ICollection<StudentWork> StudentWorks { get; set; }

    /// <summary>
    /// Gets or sets members.
    /// </summary>
    public ICollection<User>? Members { get; set; }

    /// <summary>
    /// Gets or sets criteria group.
    /// </summary>
    public required CriteriaGroup CriteriaGroup { get; set; }
}